diesel/src/pg/types/numeric.rs

```diff
@@ -112,7 +108,8 @@ mod bigdecimal {
                     .expect("enough digits exist")
                     .iter()
                     .rev()
-                    .take_while(|i| i.is_zero())
+                    .cloned()
```

Comment:
Why do we need to clone this now?

Reply:
Good question. Got an error on my first try so added cloned there. Now that error is gone if I remove it again 🤷‍♂️

Reply:
Ok, that was not that easy.
So clippy suggests to replace the closure with `Zero::is_zero`. The iterator yields `&&i16`, `Zero` is implemented for `i16`, `is_zero` takes the value by reference. So this closure works due to autoderef, whereas passing the function directly (as clippy suggests) does not work because there is no impl for `&i16`.
So there are two options here:
* Ignore that clippy warning
* Clone all values (not that expensive on a iterator over `i16`)

Reply:
I don't really have a preference as long as we open an issue with Clippy

Reply:
(Personally I hate this lint and wouldn't mind just turning it off)

Reply:
That's interesting: It's [not reproducible](https://play.rust-lang.org/?version=beta&mode=debug&edition=2018&gist=6d82809d4a8fd6552a2498bb000718ea) by simply creating an iterator over `&&i16` and it's [not reproducible by creating](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6de2bc8d499ef1b3b510f05296d4d278). Also tried locally with the a newer nightly. That does not emit this warning, so this seems to be already fixed.
Anyway, I've added a comment to [this issue](https://github.com/rust-lang/rust-clippy/issues/3942)